### PR TITLE
Adds base wifi command and tests

### LINKED ISSUE #199: "t2 wifi" alone does nothing
```
➜  t2-cli git:(master) t2 wifi
➜
```

I assume this is when there is no wifi network connected.

### PR DESCRIPTION
**Pull request by johnnyman727**
Fixes #199.

### CODE REVIEW

File: bin/tessel-2.js [hunk 1]
```diff
@@ -252,6 +234,28 @@ module.exports = function(args) {
   parser.parse(args);
 };
 
+module.exports.closeSuccessfulCommand = function() {
+  process.exit(0);
+};
+
+// Allow options to be partially applied
+module.exports.closeFailedCommand = function(opts, err) {
```
Comment (johnnyman727) [inline bin/tessel-2.js]: It's kind of ugly but I had to break `closeSuccessfulCommand` and `closeFailedCommand` into the module exports (and call them as `module.exports.X` elsewhere) to use them properly in tests. Other suggestions welcome.

File: test/unit/bin-tessel-2.js [hunk 2]
```diff
@@ -87,12 +88,11 @@ exports['Tessel (cli: restart)'] = {
   noError: function(test) {
     test.expect(1);
 
-    cli(['restart']);
+    cli(['restart', '--type=ram']);
```
Comment (johnnyman727) [inline test/unit/bin-tessel-2.js]: For some reason, the `restart` command doesn't get the default `type` property assigned to it by `nomnom` in the unit tests. The default param shows up properly w/o tests.

File: test/unit/bin-tessel-2.js [hunk 3]
```diff
@@ -239,9 +238,12 @@ exports['Tessel (cli: wifi)'] = {
   setUp: function(done) {
     this.sandbox = sinon.sandbox.create();
     this.warn = this.sandbox.stub(logs, 'warn');
+    this.info = this.sandbox.stub(logs, 'info');
     this.printAvailableNetworks = this.sandbox.stub(controller, 'printAvailableNetworks').returns(Promise.resolve());
     this.connectToNetwork = this.sandbox.stub(controller, 'connectToNetwork').returns(Promise.resolve());
-    this.processExit = this.sandbox.stub(process, 'exit');
```
Comment (johnnyman727) [inline test/unit/bin-tessel-2.js]: For some reason, `process.exit` wasn't being stubbed over properly here. The actual `process.exit` would get called and break the tests. I opted to use `closeSuccessfulCommand` insteaad.

Comment (johnnyman727) [inline test/unit/bin-tessel-2.js]: @rwaldron any idea why this might be the case?

**Comment by johnnyman727** [comment]
@MrNice could you review?

**Comment by MrNice** [comment]
@johnnyman727 I'll get it early tmw morning, sorry. Just finished a reversi app that I was assigned for a job process. https://arcane-lake-3833.herokuapp.com/ Probably the second largest job application piece I've had to do.

**Comment by johnnyman727** [comment]
@MrNice I don't really know how reversi works but the premise seems to be that you can never beat the computer.

**Comment by MrNice** [comment]
@johnnyman727 It looks good to me, but I'm not 100% clear on what the errors were. There's a few ugly things like the double `catch` that should be looked into.

Is there a Wiki Page on best practices for writing tests for the CLI? This PR opened my eyes to a few of the oddities therein.

**Comment by johnnyman727** [comment]
@MrNice The root cause appears to be that `process.exit` wasn't properly being stubbed out so the actual `process.exit` was being called which killed the test. I'm not sure why that is the case.

Where is the double `catch` that you are referring to?

We don't have documentation up for CLI tests yet. I've mostly been going off of [`nodeunit`](https://github.com/caolan/nodeunit) and [`sinon`](http://sinonjs.org/) documentation.

File: test/unit/bin-tessel-2.js [hunk 4]
```diff
@@ -285,7 +289,7 @@ exports['Tessel (cli: wifi)'] = {
     reject.catch(function() {
```
Comment (MrNice) [inline test/unit/bin-tessel-2.js]: @johnnyman727 Sorry! I was referring to these double catches over here, they are outside of the scope of the PR

Comment (johnnyman727) [inline test/unit/bin-tessel-2.js]: :+1:

**Pull request merged**